src/unix/linux.c

```diff
@@ -2308,46 +2308,70 @@ static int uv__get_cgroupv2_constrained_cpu(const char* cgroup,
                                             uv__cpu_constraint* constraint) {
   char path[256];
   char buf[1024];
-  unsigned int weight;
   int cgroup_size;
+  long long min_quota;
+  long long quota;
+  long long limit;
+  long long period;
+  char* end;
+  const char* cgroup_mount;
   const char* cgroup_trimmed;
   char quota_buf[16];
 
   if (strncmp(cgroup, "0::/", 4) != 0)
     return UV_EINVAL;
 
   /* Trim ending \n by replacing it with a 0 */
+  cgroup_mount = "/sys/fs/cgroup";
   cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
-  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first slash */
+  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first \n */
+  min_quota = LLONG_MAX;
 
-  /* Construct the path to the cpu.max file */
-  snprintf(path, sizeof(path), "/sys/fs/cgroup/%.*s/cpu.max", cgroup_size,
-           cgroup_trimmed);
+  /* Construct the path to the cpu.max files */
+  snprintf(path, sizeof(path), "%s/%.*s/cgroup.controllers", cgroup_mount,
+           cgroup_size, cgroup_trimmed);
 
-  /* Read cpu.max */
+  /* Read controllers, if not exists, not really a cgroup */
   if (uv__slurp(path, buf, sizeof(buf)) < 0)
     return UV_EIO;
 
-  if (sscanf(buf, "%15s %llu", quota_buf, &constraint->period_length) != 2)
-    return UV_EINVAL;
+  snprintf(path, sizeof(path), "%s/%.*s", cgroup_mount, cgroup_size,
+           cgroup_trimmed);
+  end = path + strlen(path);
 
-  if (strncmp(quota_buf, "max", 3) == 0)
-    constraint->quota_per_period = LLONG_MAX;
-  else if (sscanf(quota_buf, "%lld", &constraint->quota_per_period) != 1)
-    return UV_EINVAL; // conversion failed
+  /* Read cpu.max in the whole file-hierarchy */
+  while(strncmp(path, cgroup_mount, strlen(cgroup_mount)) == 0) {
+    snprintf(end, sizeof(path) - (end - path), "/cpu.max");
```

Comment:
This doesn't look right. If you modify the buffer in-place, you can't safely iterate the next lines; you may have clobbered them.

Reply:
True that, made something else